src/eliminate_identity.cpp

```diff
@@ -0,0 +1,35 @@
+#include <migraphx/eliminate_identity.hpp>
+#include <migraphx/program.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/operators.hpp>
+#include <migraphx/iterator_for.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/stringutils.hpp>
+#include <utility>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+void eliminate_identity::apply(program& p) const
+{
+    for(auto ins : iterator_for(p))
+    {
+        if(ins->get_operator().name() == "identity")
+        {
+            if(ins != p.end())
```

Comment:
This actually doesnt check if its the last instruction because `end()` is one past the last. So you need to write `ins != std::prev(p.end())`.